mmseg/models/backbones/mit.py

```diff
@@ -0,0 +1,420 @@
+import math
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import (ConvModule, Linear, build_activation_layer,
+                      build_norm_layer, constant_init, normal_init,
+                      trunc_normal_init)
+from mmcv.cnn.bricks import DropPath
+from mmcv.runner import BaseModule, ModuleList, load_checkpoint
+from torch.nn.modules.utils import _pair as to_2tuple
+
+from ...utils import get_root_logger
+from ..builder import BACKBONES
+
+
+class DWConv(BaseModule):
+
+    def __init__(self, dim=768):
+        super().__init__()
+        self.dwconv = ConvModule(
+            in_channels=dim,
+            out_channels=dim,
+            kernel_size=3,
+            stride=1,
+            padding=(3 - 1) // 2,
+            bias=True,
+            groups=dim)
+
+    def forward(self, x, H, W):
+        B, _, C = x.shape
+        x = x.transpose(1, 2).reshape(B, C, H, W)
+        x = self.dwconv(x)
+        x = x.flatten(2).transpose(1, 2).contiguous()
+
+        return x
+
+
+class Mlp(BaseModule):
+
+    def __init__(self,
+                 embed_dims,
+                 feedforward_channels,
+                 act_cfg=dict(type='GELU'),
+                 drop_rate=0.):
+        super().__init__()
+
+        in_channels = embed_dims
+        self.fc1 = Linear(in_channels, feedforward_channels)
+        self.dwconv = DWConv(feedforward_channels)
+        self.act = build_activation_layer(act_cfg)
+        self.fc2 = Linear(feedforward_channels, in_channels)
+        self.drop = nn.Dropout(drop_rate)
+
+    def forward(self, x, H, W):
+        x = self.fc1(x)
+        x = self.dwconv(x, H, W)
+        x = self.act(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+
+class Attention(BaseModule):
+
+    def __init__(self,
+                 dim,
+                 norm_cfg=dict(type='LN'),
+                 num_heads=8,
+                 qkv_bias=False,
+                 qk_scale=None,
+                 attn_drop=0.,
+                 proj_drop=0.,
+                 sr_ratio=1):
+        super().__init__()
+        assert dim % num_heads == 0, f'dim {dim} should be divided by '
+        f'num_heads {num_heads}.'
+
+        self.dim = dim
+        self.num_heads = num_heads
+        head_dim = dim // num_heads
+        self.scale = qk_scale or head_dim**-0.5
+
+        self.q = Linear(dim, dim, bias=qkv_bias)
+        self.kv = Linear(dim, dim * 2, bias=qkv_bias)
+        self.attn_drop = nn.Dropout(attn_drop)
+        self.proj = Linear(dim, dim)
+        self.proj_drop = nn.Dropout(proj_drop)
+
+        self.sr_ratio = sr_ratio
+        if sr_ratio > 1:
+            self.sr = ConvModule(
+                in_channels=dim,
+                out_channels=dim,
+                kernel_size=sr_ratio,
+                stride=sr_ratio)
+            _, self.norm = build_norm_layer(norm_cfg, dim)
```

Comment:
If we inherit from `MultiheadAttention` in MMCV, we only need to add these lines.